src/drivers/messenger/posix/driver.c

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2021 Wavious LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Standard includes. */
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+
+/* Socket APIs. */
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netinet/ip.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+/* RTOS includes. */
+#include <FreeRTOS.h>
```

Comment:
Just wondering if we want to call the directory this is in posix-freertos or something since it is strictly a posix driver for FreeRTOS specifically.  Or will we update this if we ever have to support another OS?

Reply:
You could make the same argument for the other messenger driver as well. It also uses FreeRTOS APIs. This will be taken care of by the OSAL layer.

Reply:
Ok, wasn't sure if the driver layer should be abstracted or specified which implementation it uses.  Just thinking if it will cause any potential issues, it's easier to change now if we wanted.

Reply:
No drivers should be written against the OSAL interface. They should be agnostic to OS that they're running in.